arch/arm/src/stm32f7/stm32_i2c.c

```diff
@@ -784,11 +784,13 @@ static uint32_t stm32_i2c_toticks(int msgc, struct i2c_msg_s *msgs)
       bytecount += msgs[i].length;
     }
 
-  /* Then return a number of microseconds based on a user provided scaling
-   * factor.
+  /* Then return a number of ticks based on a user provided scaling
+   * factor, rounded up.
    */
 
-  return USEC2TICK(CONFIG_STM32F7_I2C_DYNTIMEO_USECPERBYTE * bytecount);
+  return USEC2TICK(CONFIG_STM32F7_I2C_DYNTIMEO_USECPERBYTE * bytecount +
```

Comment:
Many places use the similar approach, should we update them too? (Please grep DYNTIMEO_USECPERBYTE).

Reply:
Yes, this is a wider problem. Only thing is, that I cannot test those other platforms and I don't want to touch low level driver without testing them.

I don't know how this broke in the first place, the i2c used to work fine previously, and now with the latest nuttx it didn't work pretty much at all (all smaller i2c transfers have broken). So what happened here? Has someone pushed in some untested change to i2c drivers?

Reply:
Why not be more deliberate. v != 0 ? v : 1 and not tie it to the other values.

Reply:
That makes quite large error, imagine calculation result being e.g. 1.9; would be rounded down to 1. As this is a timeout value it is better to always round up. Of course, a simple + 1 for ticks would work, not sure if I was just making it too exact in rounding up ?

Reply:
If it looks complex, we can simplify it a bit and keep it still exact; instead of " -CONFIG_STM32F7_I2C_DYNTIMEO_USECPERBYTE" in the end, just -1 works the same. I'll make this fix to prettify it. Would that do, @davids5 ?

Reply:
Yes